reference/forms/types/collection.rst

```diff
@@ -268,6 +268,13 @@ delete_empty
 
 **type**: ``Boolean`` **default**: ``false``
 
+.. caution::
+
+    Delete empty will only remove items when the normalized value is null.
+    If your `type`_ is a compound form type, you need to have the :ref:`required <reference-form-option-required>`
+    option set to ``false`` inside `options`_, See :ref:`empty_data <reference-form-option-empty-data>`.
```

Comment:
After I've added the last suggestion of `empty_data` I had to remove 'inside options' to be more stetic.